HOLMS.Messaging/RabbitConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+
+namespace HOLMS.Messaging {
+    public class RabbitConfiguration {
+        public string Host { get; protected set; }
+        public ushort Port { get; protected set; }
+        public string User { get; protected set; }
+        public string Password { get; protected set; }
+        public string VHost { get; protected set; }
+
+        public RabbitConfiguration(string connectionString) {
+            //example: amqp://user:pass@hostname:port/vhost
+            int protocolEndIndex = connectionString.IndexOf("://");
+            if (protocolEndIndex == -1) {
+                throw new ArgumentException($"No protocol specified in string {connectionString}");
+            }
+            var protocol = connectionString.Substring(0, protocolEndIndex);
+            if (protocol != "amqp" && protocol != "amqps") {
+                throw new ArgumentException($"Invalid protocol {protocol}");
+            }
+            var connection = connectionString.Substring(protocolEndIndex + "://".Length);
+            if (connection.Contains("@")) {
+                var credentials = connection.Split('@')[0].Split(':');
+                User = credentials[0];
+                if (credentials.Length > 1) {
+                    Password = credentials[1];
+                }
+                if (credentials.Length > 2) {
+                    throw new ArgumentException($"Invalid field following password: {credentials[2]}. (Note: username / password may not contain a ':'");
```

Comment:
I'm pretty sure passwords can contain a colon. I'd just tokenize on the first colon and treat the second token as the password.